src/main/scala/mesosphere/marathon/api/v2/json/V2Group.scala

```diff
@@ -36,4 +42,76 @@ object V2Group {
       groups = group.groups.map(V2Group(_)),
       dependencies = group.dependencies,
       version = group.version)
+
+  implicit val v2GroupValidator: Validator[V2Group] = validator[V2Group] { group =>
+    group.id is valid
+    group.apps is valid
+    group.groups is valid
+    group.groups is noAppsAndGroupsWithSameName(group.apps)
```

Comment:
I think there is no recursion in here. `noAppsAndGroupsWithSameName` Validator does not execute an implicit validator of type `V2Group`. It takes in `group.groups` as the `apply` argument and compares it against `group.apps`. Correct me if I am wrong.

The only reason why I build a validator like that is, that the property name `groups` could be resolved in the failure message. Instead I could do a `group is noAppsAndGroupsWithSameName` (apps can be derived from `group.apps` within the validator) and explicitly put the property name `groups` into the `RuleValidation`'s `description` property. This is the way I had it first, though automatically derived property name seemed a better choice. Should I change it to `group is noAppsAndGroupsWithSameName` and explicitly provide the property name?